tests/phpunit/api/v3/Job/RemoveexpiredgeometryentityTest.php

```diff
@@ -0,0 +1,109 @@
+    // Create a collection for states
+    $collectionParams = [
+      'label' => 'States',
+      'source' => 'ABS',
+      'geometry_collection_type_id' => $this->externalCollectionType['id'],
+    ];
+    $this->statesCollection = $this->callAPISuccess('GeometryCollection', 'create', $collectionParams);
+    // Create a geometry type for state geometries
+    $geometryTypeParams = [
+      'label' => 'State',
+    ];
+    $this->stateGeometryType = $this->callAPISuccess('GeometryType', 'create', $geometryTypeParams);
+    // Create a collection for SA1 Geometries
+    $sa1CollectionParams = [
+      'label' => 'SA1',
+      'source' => 'ABS',
+      'geometry_collection_type_id' => $this->externalCollectionType['id'],
+    ];
+    $this->sa1Collection = $this->callAPISuccess('GeometryCollection', 'create', $sa1CollectionParams);
+    // Create a Geometry Type for SA1 Geometry
+    $sa1GeometryTypeParams = [
+      'label' => 'Stastical Area Level 1',
+    ];
+    $this->sa1GeometryType = $this->callAPISuccess('GeometryType', 'create', $sa1GeometryTypeParams);
+    parent::setUp();
+  }
+
+  public function tearDown() {
+    $this->callAPISuccess('GeometryType', 'delete', ['id' => $this->stateGeometryType['id']]);
+    $this->callAPISuccess('GeometryCollection', 'delete', ['id' => $this->statesCollection['id']]);
+    $this->callAPISuccess('GeometryType', 'delete', ['id' => $this->sa1GeometryType['id']]);
+    $this->callAPISuccess('GeometryCollection', 'delete', ['id' => $this->sa1Collection['id']]);
+    $this->callAPISuccess('GeometryCollectionType', 'delete', ['id' => $this->externalCollectionType['id']]);
+    parent::tearDown();
+  }
+
+  /**
+   * Simple example test case.
+   *
+   * Note how the function name begins with the word "test".
+   */
+  public function testExpiredJob() {
+    // Load geoJSON file and create a geometry
+    $sa1JSON = file_get_contents(\CRM_Utils_File::addTrailingSlash($this->jsonDirectoryStore) . 'sample_sa1_geometry.json');
+    // Create SA1 Geometry
+    $sa1 = $this->callAPISuccess('Geometry', 'create', [
+      'label' => '1210113836',
+      'geometry_type_id' => $this->sa1GeometryType['id'],
+      'collection_id' => [$this->sa1Collection['id']],
+      'geometry' => trim($sa1JSON),
+    ]);
+    $contact1 = $this->individualCreate();
+    $contact2 = $this->individualCreate();
+    $this->callAPISuccess('Geometry', 'createentity', [
+      'entity_id' => $contact1,
+      'entity_table' => 'civicrm_contact',
+      'geometry_id' => $sa1['id'],
+    ]);
+    $this->callAPISuccess('Geometry', 'createentity', [
+      'entity_id' => $contact2,
+      'entity_table' => 'civicrm_contact',
+      'geometry_id' => $sa1['id'],
+      'expiry_date' => date('Y-m-d H:m:s', strtotime('-1 month')),
+    ]);
+    $currentGeometryEntities = $this->callAPISuccess('Geometry', 'getentity', ['geometry_id' => $sa1['id']]);
```

Comment:
This looks right to me, but just to confirm I'm understanding it correctly: You are testing two things here: Both that it successfully removes an expired relationship/placement and that it does not remove one that has a null/unset expiry date, correct?

Reply:
You're spot on @unfinishedteleporter. @seamuslee001 and I discussed the concept of expiry for geom-entity relationships. Figured the default is no expiry, so any job that looks to remove expired ones should respect this fact.